### DISCUSSION TITLE
Arch Linux AUR installer.

### DISCUSSION THREAD

**Discussion by satiricon** [Difegue/LANraragi#333] [Show and tell]
I don't know if this is the right way to ask for a documentation change. If not, please delete this.

I created a Arch Linux build package for LRR. [Link](https://aur.archlinux.org/packages/lanraragi/).

Using the AUR package the installation process in Arch Linux would be as easy as entering `pikaur -S lanraragi` in the command line if the user is using pikaur. Using other AUR managers should be just as easy.

If the user is not using a AUR package manager the installation process would be something like:

```
wget https://aur.archlinux.org/cgit/aur.git/snapshot/lanraragi.tar.gz   -O - | tar -xz
cd lanraragi
makepkg -rsi
```
That would take care of installing LRR together with build and normal dependencies and deleting build dependencies after successful building and installing.

The installer also creates a lanraragi.service unit file for starting, restarting and stopping LRR with systemd's `systemctl`. If Redis is down it will get it up.

ej: `systemctl start lanraragi.service`
`systemctl restart lanraragi.service`
`systemctl stop lanraragi.service`
`systemctl status lanraragi.service`

This also gives an easy way to read the log if necessary.

`journalctl -u lanraragi -f`

**Comment by Difegue** [comment]
Wow, this is great! It ain't a real open source project until someone stuffed it into the AUR.  

I'd recommend making a pull request for the [Documentation](https://github.com/Difegue/LANraragi/tree/dev/tools/Documentation) to integrate your changes. That, or I can just copy-paste the text of this issue directly as it's clear enough...

**Comment by satiricon** [comment]
I'll make the pull request. Should I add another installation section in `methods.md`? Or a subsection in `source.md`?

**Comment by Difegue** [comment]
I've been thinking about a new installation section - If it's only about the AUR package for now it's fine but I might extend it into a "community" installation methods section later on.

It'd give me an excuse to mention the existing unraid community package as well!

**Comment by ghost** [comment]
@satiricon It gives me this error

```
Can't load application from file "/usr/lib/lanraragi/script/lanraragi": Can't locate local/lib.pm in @INC (you may need to install the local::lib module) (@INC contains: /usr/lib/lanraragi/script/../lib /usr/lib/perl5/5.32/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/5.32/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/5.32/core_perl /usr/share/perl5/core_perl) at /usr/lib/lanraragi/script/../lib/LANraragi.pm line 3.
BEGIN failed--compilation aborted at /usr/lib/lanraragi/script/../lib/LANraragi.pm line 3.
Compilation failed in require at (eval 102) line 1.
Compilation failed in require at (eval 101) line 1.
```

Also it creates perl5 directory with libraries inside $HOME during build process.

**Reply by satiricon** [reply]
Can you give me more info in order to duplicate the issue? How are you trying to install the package? 

You might be not installing make dependencies? If you are using makepkg you might want to use the `-s` or `--syncdeps` option.

**Reply by ghost** [reply]
I can build and install the package (I tried yay and then makepkg), but it crashes with this error when I run it. It doesn't matter if I run it using systemd or manually with `perl /usr/lib/lanraragi/script/launcher.pl -f /usr/lib/lanraragi/script/lanraragi` the error is the same.

**Reply by ghost** [reply]
Also here's part of the output during build process.

```
Installing Perl modules... This might take a while.

Installing Linux::Inotify2 for non-macOS systems...
Linux::Inotify2 not installed! Trying to install now using cpanm
!
! Can't write to /usr/share/perl5/site_perl and /usr/bin: Installing modules to /home/user/perl5
! To turn off this warning, you have to do one of the following:
!   - run me as a root or with --sudo option (to install to /usr/share/perl5/site_perl and /usr/bin)
!   - Configure local::lib in your existing shell to set PERL_MM_OPT etc.
!   - Install local::lib by running the following commands
!
!         cpanm --local-lib=~/perl5 local::lib && eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
!
Linux::Inotify2 is up to date. (2.2)
!
! Can't write to /usr/share/perl5/site_perl and /usr/bin: Installing modules to /home/user/perl5
! To turn off this warning, you have to do one of the following:
!   - run me as a root or with --sudo option (to install to /usr/share/perl5/site_perl and /usr/bin)
!   - Configure local::lib in your existing shell to set PERL_MM_OPT etc.
!   - Install local::lib by running the following commands
!
!         cpanm --local-lib=~/perl5 local::lib && eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
!
--> Working on ./tools/.
```

**Reply by satiricon** [reply]
Awesome I think I know what's the problem. Let me check.

**Reply by satiricon** [reply]
@Svadkos I pushed a new PKGBUILD to the AUR adding a dependency for perl-local-lib. Pretty sure that was the problem as I was able to duplicate the problem and fixed it with that change but there might be more unseen (by me) dependencies. Can you check if it works now?

Also it might be better to post issues about the installer in the AUR itself or in the installer repo https://github.com/satiricon/lanraragi-aur.

Cheers!
